cluster-autoscaler/utils/taints/taints.go

```diff
@@ -416,3 +442,50 @@ func FilterOutNodesWithStartupTaints(taintConfig TaintConfig, allNodes, readyNod
 	}
 	return newAllNodes, newReadyNodes
 }
+
+// CountNodeTaints counts used node taints.
+func CountNodeTaints(nodes []*apiv1.Node, taintConfig TaintConfig) map[string]int {
+	foundTaintsCount := make(map[string]int)
+	for _, node := range nodes {
+		for _, taint := range node.Spec.Taints {
+			key := getTaintTypeToReport(taint.Key, taintConfig)
+			if _, ok := foundTaintsCount[key]; ok {
```

Comment:
Missing map entries will default to 0, so there's no need to handle "not found" case separately - simple increment will suffice.

Reply:
Done